jhack/charm/provision.py

```diff
@@ -163,8 +164,12 @@ def _provision(
     status = juju_status(json=True)
     targets = tuple(_get_provisioner_targets(target, status))
 
+    check_destructive_commands_allowed(
+        "provision", f"would run {tf_script} on {targets}"
+    )
+
     if dry_run:
-        print(f"[dry run]: with script: {tf_script}")
+        print(f"would run: {tf_script}")
```

Comment:
I'm curios about the difference in message between destructive commands check and dry run, for one the targets are shown and for the others they aren't? I was guessing that dry run would be quite similar to the prompt? As in, in the new version, the dry run output would be shown and then ask for a prompt to continue?

Reply:
dry_run will also print the targets, a few lines below.
In most other commands I managed to add the check after the dry-run return path to make sure there's no duplication, but in this case there were too many paths.